### PR TITLE
MAINT - Update release docs and add issue template

### LINKED ISSUE #585: MAINT - Add new issue templates
@trallard - (Non-blocking comment) Shall we add a new issue template for releases and link to that in the docs here?

_Originally posted by @pavithraes in https://github.com/conda-incubator/conda-store/pull/580#discussion_r1329111984_

### PR DESCRIPTION
**Pull request by trallard**
Fixes #585 

## Description

This pull request:

- Updates the release docs
- Adds a release issue template

## Pull request checklist

- [ ] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

**Comment by pavithraes** [comment]
@trallard - Thanks for this PR! Currently, the release issue template is an empty file - did you mean to move over the checklist from the release process documentation?

**Comment by trallard** [comment]
@pavithraes  of course I forgot to push the last commit 🤦🏽‍♀️  sorted now

File: .github/ISSUE_TEMPLATE/new-release.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    hatch build
+    twine check dist/*
+    hatch clean
+    ```
+
+- [ ] Make a release commit: ``git commit -m 'REL - 2023.9.1'``
+- [ ] Push the release (REL) commit ``git push upstream main``
+- [ ] If a **release candidate** is needed, tick this box when we're ready for a full release.
+
+### 2. Make the release
```
Comment (pavithraes) [inline .github/ISSUE_TEMPLATE/new-release.md:47]:
```suggestion
### 3. Make the release
```

File: docusaurus-docs/community/release.md [hunk 2]
```diff
@@ -71,40 +71,52 @@ Release captain responsible - <@gh_username>
...
+    twine check dist/*
+    hatch clean
+    ```
+
 - [ ] Make a release commit: ``git commit -m 'REL - 2023.9.1'``
 - [ ] Push the release (REL) commit ``git push upstream main``
 - [ ] If a **release candidate** is needed, tick this box when we're ready for a full release.
 
-**Make the release**
+### 2. Make the release
```
Comment (pavithraes) [inline docusaurus-docs/community/release.md:102]:
```suggestion
### 3. Make the release
```

**Review by pavithraes — APPROVED** [review]
Thank you! This looks good to me!

Approving because the suggestions are just tiny typo-fixes. :)

**Pull request merged**